### ISSUE TITLE
Generate custom mavlink messages

### ISSUE THREAD

**Issue by Davidsastresas** [HD-Fpv/Open.HD_Image_Builder#47]
Hello, I have my own custom version of ardupilot and qgroundcontrol with custom mavlink messages. I want to add my new mavlink messages to the mavlink understood by openhd, so openhd can pass the new messages too.

**Comment by pilotnbr1** [comment]
Hi David, thanks for your interest!

The builder (this repo) pulls from a separate repo where we keep all the working bits of openhd. Specifically the Mavlink stuff is here-

https://github.com/HD-Fpv/Open.HD/tree/master/wifibroadcast-osd

**Comment by pilotnbr1** [comment]
If you are intending to just do a mod on your own image, I would suggest doing it locally on the image you have created. If you have something special that you think might benefit others, we can discuss adding your own branch to the repo...

**Comment by Davidsastresas** [comment]
Thank you very much. Indeed they are mavlink messages for a particular project I have, I don't think they have interest at all for a general ardupilot build. Thanks

Best regards

**Comment by pilotnbr1** [comment]
One unusual note regarding work on the osd is that it gets compiled on the groundpi upon every startup... Hope that saves you time!

**Comment by Davidsastresas** [comment]
That is really good to know! thank you. Although I think I need to build the project anyway myself, because the work that I need to be done is over mavlink directly, as I am adding new message definitions. I will probably ask some more question at some point.

Thank you very much for your time.

Best regards.

**Comment by pilotnbr1** [comment]
Yes, I believe the mavlink headers are compiled upon building the image (they dont recompile with the osd)

**Comment by HD-Fpv** [comment]
closing issue so that all issues are tracked on the main openHD repo..
